### PR TITLE
Feature/tao 3402 associate bluprint section

### PR DESCRIPTION
**Pull request by antoinerobin**
needs : https://github.com/oat-sa/tao-core/pull/1091

### CODE REVIEW

File: views/js/controller/creator/helpers/sectionBlueprints.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * @returns {object}
+     */
+    function getBlueprint(getUrl, model){
+
+        return $.ajax({
+            url: getUrl,
+            type: 'POST',
+            data: {
+                section: model.identifier,
+                test: 'http://tao.local/mytao.rdf#i148058656329575'
```
Comment (krampstudio) [inline views/js/controller/creator/helpers/sectionBlueprints.js]: Are you sure ?

File: views/js/controller/creator/views/section.js [hunk 2]
```diff
@@ -342,6 +347,75 @@ function($, _, uri, __, actions, itemRefView, rubricBlockView, templates, qtiTes
...
+            });
+
+            /**
+             * Start the blueprint editing
+             * @private
+             */
+            function initBlueprint(){
+
+                if(model.blueprint === undefined){
+                    var promise = sectionBlueprint.getBlueprint(data.routes.blueprintByTestSection, model);
```
Comment (krampstudio) [inline views/js/controller/creator/views/section.js]: 2 smalls issues : 
 - `var`declaration should be done at the top of the scope to reflect hoisting
 - don't call it a promise, it's a defered.

But you can just chain the call
```js
sectionBlueprint
  .getBlueprint(data.routes.blueprintByTestSection, model)
  .success(...
```

File: actions/class.Creator.php [hunk 3]
```diff
@@ -47,6 +47,11 @@ public function index(){
             $this->setData('saveUrl', _url('saveTest', null, null, array('uri' => $testUri)));
             $this->setData('itemsUrl', _url('get', 'Items'));
             $this->setData('categoriesUrl', _url('getCategories', 'Items'));
+
+            if(common_ext_ExtensionsManager::singleton()->isInstalled('taoBlueprints')){
```
Comment (krampstudio) [inline actions/class.Creator.php]: Is it allowed ?

File: views/js/controller/creator/helpers/sectionBlueprints.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
```
Comment (krampstudio) [inline views/js/controller/creator/helpers/sectionBlueprints.js]: 2015

File: actions/class.Creator.php (hunk 3 above)
Comment (antoinerobin) [inline actions/class.Creator.php]: yes, it is the only way to do it

**Comment by krampstudio** [comment]
Without blueprints defined and searching in the field results in an error : 
`TypeError: g.results is undefined` 
This is because the response of the ajax call is empty. You should have an empty object/array

**Comment by antoinerobin** [comment]
https://github.com/oat-sa/extension-tao-blueprints/pull/7 should fix your issue

**Pull request merged**